core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/LazyHttpClient.java

```diff
@@ -118,12 +109,20 @@ private static CloseableHttpClient init() {
                         .build());
         cm.setMaxTotal(DEFAULT_MAX_TOTAL_CONNECTIONS);
         cm.setDefaultMaxPerRoute(DEFAULT_MAX_CONNECTIONS_PER_ROUTE);
+        cm.setDefaultSocketConfig(SocketConfig.custom()
+                .setSoTimeout(REQUEST_TIMEOUT_IN_MILLIS)
```

Comment:
what is setSoTimeout?

Reply:
socket timeout, from reading a bunch of Apache HttpClient jira tickets yesterday (😭), I think it's used for SSL connection/handshake, which occurs outside of "request" scope, so outside of the request-level timeouts we set later in this method